iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs

```diff
@@ -413,8 +413,7 @@ public override async Task SendTwinPatchAsync(TwinCollection reportedProperties,
                 }
                 else
                 {
-                    // Timeout happen
-                    throw new TimeoutException();
+                    throw new IotHubClientException("Timeout happen before task completes.", true, IotHubStatusCode.Timeout);
```

Comment:
This error message doesn't really explain much. Did the client time out waiting for the service to send the twin response here?

Reply:
Yes, it should be. I will explain with more details in the error message.